pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -951,9 +951,20 @@ func (gce *GCECloud) firewallObject(name, region, desc string, sourceRanges nets
 	for ix := range ports {
 		allowedPorts[ix] = strconv.Itoa(int(ports[ix].Port))
 	}
-	hostTags, err := gce.computeHostTags(hosts)
-	if err != nil {
-		return nil, err
+	// If the node tags to be used for this cluster have been predefined in the
+	// provider config, just use them. Otherwise, grab all tags from all relevant
+	// instances:
+	hostTags := gce.nodeTags
+	if len(hostTags) == 0 {
+		if hostTagsTmp, err := gce.computeHostTags(hosts); err != nil {
+			return nil, err
+		} else {
+			hostTags = hostTagsTmp
+		}
+	}
+	if len(hostTags) == 0 {
+		return nil, fmt.Errorf("No node tags supplied and failed to parse the given" +
```

Comment:
I don't think we use "+" anywhere (you can leave this in if I'm wrong). If I'm right, please make this conform to the other places where we either put everything on a single line, or split the entire string starting at "No" node onto the new line and leave a hanging "(", i.e:
```
Errorf(
    "test")
```